.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,38 @@
+name: PR Patchset
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: Tests using ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        experimental: [false]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Check Python Version
+      run: |
+        python --version
+
+    - name: Install Dependencies Required for Tests to be run
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox tox-gh-actions
+
+    - name: Run Tests
+      run: tox || true
```

Comment:
Rather than running everything here, maybe we should run each of the tests separately so that it is easy at a glance to see what failed.

Reply:
I am good with that. You mean run one tox env Per run and run it over n matrix ? I was actually considering splitting the test tasks and sanity check like type check and linter to a different file so that I can easily manage them. Would that be ok ?

Reply:
You mean a different `tox.ini`?

What I meant was having something like this:

```
- name: Run Tests 
  run: tox -e py38
- name: Run Tests 
  run: tox -e py38
- name: Type checking
  run: tox -e type-checking
...  
```

Reply:
Then there is not a ridiculously  long list of checks on a PR, but when you go into a specific workflow, it is easy to drilldown and see (for example) that it was a linting issue that caused a failure (instead of needing to look through the entire `tox` output.